gradle/libs.versions.toml

```diff
@@ -3,14 +3,17 @@
 feature-flag = "2.4.0"
 
 # https://developer.android.com/studio/releases/gradle-plugin
-android-gradle = "4.2.2"
+android-gradle = "8.1.3"
 
 # https://kotlinlang.org/docs/releases.html#release-details
-kotlin = "1.4.31"
+kotlin = "1.9.10"
```

Comment:
AGP 8.1 is using kotlin 1.8.20
https://android.googlesource.com/platform/tools/buildSrc/+/refs/tags/studio-2022.3.1/base/dependencies.properties#108

So, We should use same kotlin version with AGP.

Reply:
Gradle 8.4 uses Kotlin 1.9.10 and it seems the gradle's version is used even if we specify 1.8.20 manually.
https://docs.gradle.org/8.4/release-notes.html
So I'll downgrade Gradle to 8.2.1 which uses Kotlin 1.8.20.

Reply:
> Gradle 8.4 uses Kotlin 1.9.10 

This means that Gradle 8.4 is using kotlin 1.9.10 to compiling `kotlin dsl script` and `gradle kotlin dsl library`.
And AGP 8.1 is using [gradle tooling api 8.0](https://android.googlesource.com/platform/tools/buildSrc/+/refs/tags/studio-2022.3.1/base/dependencies.properties#103) which is minimum requirements of gradle version.

In our in-house cases, we try to use the latest toolchain.
But in the world of Github, not all devs use only the latest toolchain.
Therefore, to keep a reasonable user pool, it is recommended to use an slightly older version of Gradle.
In my opinion, Gradle 8.2.1 seems appropriate since it has been released for half a year.